prereceivecli/lib/utils.py

```diff
@@ -209,32 +209,12 @@ def get_table_for_project_group(project_group, credentials):
     return table
 
 
-@contextmanager
-def no_quarantine():
-    """Context manager that clears the GIT_QUARANTINE_PATH environment variable and restores it."""
-    git_quarantine_path = os.environ.get('GIT_QUARANTINE_PATH')
-    try:
-        if git_quarantine_path is not None:
-            del os.environ['GIT_QUARANTINE_PATH']
-        yield
-    finally:
-        if git_quarantine_path is not None:
-            os.environ['GIT_QUARANTINE_PATH'] = git_quarantine_path
-
-
-class GitCheckout:
+class HashChecker:
     """Implements a git rebuilding context manager for a pre-receive hook."""
 
-    def __init__(self, project, hasher, entries):
-        logger_name = u'{base}.{suffix}'.format(base=LOGGER_BASENAME,
-                                                suffix=self.__class__.__name__)
-        self._logger = logging.getLogger(logger_name)
-        self.project = project
-        self.hasher = hasher
-        self.entries = entries
-        self.temporary_directory = None
-        self.git_directory = None
-        self.working_directory = None
+    def __init__(self):
+        self._logger = logging.getLogger(f'{LOGGER_BASENAME}.{self.__class__.__name__}')
+        self.hasher = Hasher()
```

Comment:
I guess we are only using this in the context of the validation so maybe use it locally there and not have it as an object attribute?

Reply:
Makes sense, done.